airflow/kubernetes/pod_generator.py

```diff
@@ -453,7 +453,7 @@ def make_unique_pod_id(pod_id: str) -> str:
         # Strip trailing '-' and '.' as they can't be followed by '.'
         trimmed_pod_id = pod_id[:MAX_LABEL_LEN].rstrip('-.')
 
-        safe_pod_id = f"{trimmed_pod_id}.{safe_uuid}"
+        safe_pod_id = f"{trimmed_pod_id}-{safe_uuid}"
```

Comment:
In the case that the pod_id is long enough, then there will be an error even with the trimming introduced to prevent this  errorfrom occurring. This is because the safe_pod_id name appends the safe_uuid after trimming, which would make the returned label greater than the allowable label length MAX_LABEL_LEN.

Should the trimming occur after the safe pod id is constructed?

Reply:
Valid point, but we also can't trim construction, since it might cause collisions. Since we ensure that trimmed_pod_id is limited, all we need to ensure is that uuid is limited too. Now, knowing that safe_uuid is static length, we can calculate the exact leftover space for trimmed_pod_id.

In other words, if length is the initial problem of nearby ticket, I would go with something like:
```
trimmed_pod_id = pod_id[:MAX_LABEL_LEN-len(safe_uuid)].rstrip('-.')
```

Reply:
This feels like the right solution to me. Could you update the PR to use this?

A comment explaining why the various cleanups are needed would also be nice.